### ISSUE TITLE
How do I write null values?

### ISSUE THREAD

**Issue by derekperkins** [ClickHouse/clickhouse-go#124]
I'm trying to insert null values, so I have columns like `*int32` and nested columns like `[]*int16`, but that results in an error like this.
`FieldName (Int32): unexpected type *int32`

Is there a different supported way to write nil values? I do this fairly often with the `mysql` driver.

**Comment by derekperkins** [comment]
Should all the `/lib/column/` Write methods add type assertions for the corresponding pointer values?

https://github.com/kshvakov/clickhouse/blob/aeedb7b9d0584f393905447021238b5c6e9c5154/lib/column/int32.go#L18-L29

**Comment by derekperkins** [comment]
I don't think that test covers typical usage. You are testing inserting a nil constant, where normal usage is going to be `*int32`, `*string`, etc, that may or may not be null.

https://github.com/kshvakov/clickhouse/blob/master/clickhouse_nullable_test.go#L213-L230

**Comment by derekperkins** [comment]
This line isn't ever going to return nil except for a nil literal. https://github.com/kshvakov/clickhouse/blob/master/lib/column/nullable.go#L54

See example: https://play.golang.org/p/QuyrmlpqX03

**Comment by derekperkins** [comment]
I think the two options are to add reflection to that if statement or to add checks on all the type switches. I think I'd prefer the type switching option, but the flow of the program makes it easier to do the first.

**Comment by dstruck** [comment]
> Nullable example https://github.com/kshvakov/clickhouse/blob/master/clickhouse_nullable_test.go#L14
> But I'm not sure that `Nullable(Array(*))`is working correctly, I'll check it.

Nullable support for arrays does not seem to work yet, at least for the DateTime type.

**Default DateTime array:**
```
# clickhouse-client
CREATE TABLE test (ts Array(DateTime)) ENGINE=Memory;
INSERT INTO test VALUES (array(now(), now()));

# go client
_, err = stmt.Exec(clickhouse.ArrayDateTime([]time.Time{time.Now(), time.Now()}))
```
Works.

**Nullable DateTime array:**
```
# clickhouse-client
CREATE TABLE test (ts Array(Nullable(DateTime))) ENGINE=Memory;
INSERT INTO test VALUES (array(now(), NULL));
```
Works.
```
# go client
_, err = stmt.Exec(clickhouse.ArrayDateTime([]time.Time{time.Now(), time.Now()}))
```
Does not work and gives back the following error message:

> error commiting: code: 101, message: Unexpected packet Hello received from client
> exit status 1

**Comment by kshvakov** [comment]
I'll add support for the Nullable(Array) in the near feature.

**Comment by vicdashkov** [comment]
Would you please add support for IPv4 and IPv6? Thanks!

**Comment by rolinh** [comment]
@vicdashkov This is addressed in #352 ;)
